ecs-cli/modules/config/config_v1.go

```diff
@@ -84,73 +84,84 @@ func NewCLIConfig(cluster string) *CLIConfig {
-//  3) AWS Profile - attempts to use credentials (aws_access_key_id, aws_secret_access_key) or assume_role (role_arn, source_profile) from AWS profile name
-//    a) profile name from ECS config file (OR)
-//    b) AWS_PROFILE environment variable (OR)
-//    c) AWS_DEFAULT_PROFILE environment variable (defaults to 'default')
-//  4) EC2 Instance role
-func (cfg *CLIConfig) ToAWSSession() (*session.Session, error) {
-	svcConfig := aws.Config{}
-	return cfg.toAWSSessionWithConfig(svcConfig)
+//  1) ECS CLI Profile Flags
+//   a) ECS Profile (--ecs-profile)
+//   b) AWS Profile (--aws-profile)
+//  2) Environment Variables - attempts to fetch the credentials from environment variables:
+//   a) ECS_PROFILE
+//   b) AWS_PROFILE
+//   c) AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY, Optional: AWS_SESSION_TOKEN
+//  3) ECS Config - attempts to fetch the credentials from the default ECS Profile
+//  4) Default AWS Profile - attempts to use credentials (aws_access_key_id, aws_secret_access_key) or assume_role (role_arn, source_profile) from AWS profile name
+//    a) AWS_DEFAULT_PROFILE environment variable (defaults to 'default')
+//  5) EC2 Instance role
+func (cfg *CLIConfig) ToAWSSession(context *cli.Context) (*session.Session, error) {
+
+	region := cfg.getRegion()
+
+	if isProfileFlagsCase(context) {
+		return credsFromECSConfig(cfg)
+	} else if isEnvVarCase(context) {
+		return defaultProvider(region)
+	} else if isDefaultECSProfileCase(cfg) {
+		return credsFromECSConfig(cfg)
+	} else {
+		return defaultProvider(region)
+	}
+
 }
 
-func (cfg *CLIConfig) toAWSSessionWithConfig(svcConfig aws.Config) (*session.Session, error) {
-	credentialProviders := cfg.getInitialCredentialProviders()
-	chainCredentials := credentials.NewChainCredentials(credentialProviders)
-	if _, err := chainCredentials.Get(); err == nil {
-		svcConfig.Credentials = chainCredentials
-	}
+func isProfileFlagsCase(context *cli.Context) bool {
+	return (recursiveFlagSearch(context, command.ECSProfileFlag) != "" || recursiveFlagSearch(context, command.AWSProfileNameFlag) != "")
+}
 
-	svcConfig.Region = aws.String(cfg.getRegion())
+func isEnvVarCase(context *cli.Context) bool {
+	return (os.Getenv(command.AWSSecretKeyEnvVar) != "" && os.Getenv(command.AWSAccessKeyEnvVar) != "")
+}
 
-	svcSession, err := session.NewSessionWithOptions(session.Options{
-		Config:            svcConfig,
-		Profile:           cfg.AWSProfile,
-		SharedConfigState: session.SharedConfigEnable,
-	})
-	if err != nil {
-		return nil, err
-	}
-	region := *svcSession.Config.Region
-	if region == "" {
-		return nil, fmt.Errorf("Set a region using ecs-cli configure command with the --%s flag or %s environment variable or --%s flag", cli.RegionFlag, cli.AwsRegionEnvVar, cli.ProfileFlag)
+func isDefaultECSProfileCase(cfg *CLIConfig) bool {
+	return (cfg.AWSAccessKey != "" || cfg.AWSSecretKey != "" || cfg.AWSProfile != "")
+}
+
+func credsFromECSConfig(cfg *CLIConfig) (*session.Session, error) {
```

Comment:
The naming here is a bit confusing, since I'd expect this function to return `creds` but it's returning a `Session` with those creds. It's also not totally clear to me that, based on where it's called in line 115, how this follows the ordering contract of ECS creds first, then AWS creds.